### DISCUSSION TITLE
Ability to import and/or arrange objects into unused space within the current bounds

### DISCUSSION THREAD

**Discussion by Darantius** [sn4k3/UVtools#832] [Ideas]
Currently, importing a new object will always put it outside the current green bounds rectangle. This makes printing a variety of different models tedious as you need to strategically combine files in certain orders to use the plate space effectively, or know exactly which location 

It would save so much time if the system could recognise there is enough empty space inside the current green bounds to fit the importing file. Alternatively, if each object kept its own bounds and you could choose how they were arranged.

**Comment by sn4k3** [comment]
You referring to the Stack import method right? It stacks models by it bounding rectangle, putting one next to the other. Optimize and nest per curves is not easy.

One alternative is by use the MergeMax method, so it gain the position of the model you are importing, but require correct position to not colide with the current.

After stack is also possible to move one region, by using the Tool Move and Roi selection, but requires try/error with Ctrl + Shift + Z

**Comment by Hazoooka** [comment]
> It stacks models by it bounding rectangle, putting one next to the other

It also automatically moves the existing models making room for the new ones.
Stack import option is what you are looking for Dariantius.
